x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -202,10 +202,18 @@ export const parseModelStateFromStats = (
     modelTypes?.includes(TRAINED_MODEL_TYPE.LANG_IDENT)
   )
     return TrainedModelState.Started;
-  switch (model?.deployment_stats?.state) {
+
+  return parseModelState(model?.deployment_stats?.state)
+};
+
+export const parseModelState = (state?: string) => {
```

Comment:
These new states will never actually show up on `deployment_stats` right? 

Just wondering if we need to worry about adding any tests for them since `parseModelStateFromStats` does have unit tests currently.

Reply:
Correct, `parseModelState()` is more generic and works with a string. Should we add tests for `parseModelState()` with the various state possibilities?

Reply:
ideally yes, should be easy to unit test.

Reply:
Added tests.